extensions/utm-tracker/src/index.ts

```diff
@@ -0,0 +1,59 @@
+import { register } from '@shopify/web-pixels-extension';
+
+type utmTracker = {
+   utm_source?: string;
+   utm_medium?: string;
+   utm_campaign?: string;
+   utm_content?: string;
+   timestamp: number;
+};
+register(({ analytics, browser, init, settings }): void => {
+   const UTM_TRACKERS_KEY = 'utmTrackers';
+
+   const storeUtmTrackers = (): void => {
+      const url = new URL(init.context.document.location.href);
+      const utmTrackers = url.searchParams.get(UTM_TRACKERS_KEY);
+      if (utmTrackers) {
+         void browser.localStorage
+            .setItem(UTM_TRACKERS_KEY, utmTrackers)
+            .catch((error) => console.error(error));
+      } else {
+         void browser.localStorage
+            .removeItem(UTM_TRACKERS_KEY)
+            .catch((error) => console.error(error));
```

Comment:
It doesn't seem like any of the `void` operators in this file are necessary. Did you add them to suppress an eslint warning or something? I think it would be more clear to make this an async function and use `await` to indicate that they return promises.

```suggestion
         await browser.localStorage
            .setItem(UTM_TRACKERS_KEY, utmTrackers)
            .catch((error) => console.error(error));
      } else {
         await browser.localStorage
            .removeItem(UTM_TRACKERS_KEY)
            .catch((error) => console.error(error));
```